src/auth.ts

```diff
@@ -1,16 +1,70 @@
+import { data, redirect } from '@remix-run/node';
 import { getAuthorizationUrl } from './get-authorization-url.js';
-import { terminateSession } from './session.js';
+import { refreshSession, terminateSession } from './session.js';
 
-async function getSignInUrl(returnPathname?: string) {
+export async function getSignInUrl(returnPathname?: string) {
   return getAuthorizationUrl({ returnPathname, screenHint: 'sign-in' });
 }
 
-async function getSignUpUrl(returnPathname?: string) {
+export async function getSignUpUrl(returnPathname?: string) {
   return getAuthorizationUrl({ returnPathname, screenHint: 'sign-up' });
 }
 
-async function signOut(request: Request) {
+export async function signOut(request: Request) {
   return await terminateSession(request);
 }
 
-export { getSignInUrl, getSignUpUrl, signOut };
+/**
+ * Switches the current session to a different organization.
+ * @param request - The incoming request object.
+ * @param organizationId - The ID of the organization to switch to.
+ * @param options - Optional parameters.
+ * @returns A redirect response to the specified returnTo URL or a data response with the updated auth data.
+ */
+export async function switchToOrganization(
+  request: Request,
+  organizationId: string,
+  { returnTo }: { returnTo?: string } = {},
+) {
+  try {
+    const auth = await refreshSession(request, { organizationId });
+
+    // if returnTo is provided, redirect to there
+    if (returnTo) {
+      return redirect(returnTo, {
+        headers: {
+          'Set-Cookie': auth.headers?.['Set-Cookie'] ?? '',
+        },
+      });
+    }
+
+    // otherwise return the updated auth data
+    return data(
+      { success: true, auth },
+      {
+        headers: {
+          'Set-Cookie': auth.headers?.['Set-Cookie'] ?? '',
+        },
+      },
+    );
+  } catch (error) {
+    if (error instanceof Response && error.status === 302) {
+      throw error;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const errorCause: any = error instanceof Error ? error.cause : null;
```

Comment:
Could we throw a more specific error type that has cause on it so we don't have to use any?

Reply:
<img width="1148" alt="image" src="https://github.com/user-attachments/assets/9db93410-83fc-4e65-9fb3-df9e79381e40" />

The error will always be `unknown` or `any`, and IMO any additional type checks are unnecessary as this is isolated to just this line of checks.